Controller/item controller

Mavo39 opened the pull request:
[Controller/ItemController] Add and Edit some files
Changes:

Add some routes to use "Add" and "Edit" item functions
Add ItemController's functions to use
Edit other member's code to add my routes and adjust my code to everyone
Delete my seederfile to accept member's one

Mavo39 committed:
[Controller/ItemController] - Add a new CSS and Edit some files
Changes:

Add
 - delete route on item
 - include on my_items.blade.php to use modal
 - route of add item
 - item.css to adjust on add and edit item page
 - categories function on user model
 - categories and user functions on item model
 - some functions on ItemController and DonationController
 - $item on CategoryController to solve a error
Edit
 - add and edit blade.php
 - donations/delete modal to use components
 - donations/show.blade.php
 - components delete modal to show letters at middle
 - style.css to organize
Delete
 - items/show.blade.php because it's temporary until my_item page creates
 - me-2 on components btn.blade.php
Move dm.blade.php under users folder

Mavo39 committed:
Merge branch 'main' into Controller/ItemController

Mavo39 committed:
[Controller/ItemController] - Add and Edit some files
Changes:

Add
 - destroy function on DonationController
 - "users." on HomeController to jump to homepage
 - relationship between user and category
 - DonationItemSeeder to confirm my part
 - ItemSeeder to confirm my part
 - photo whether the photo works
 - donations/item.edit page
 - edit item route on my_item.blade.php
Edit
 - functions on ItemController
 - CategorySeeder to confirm my part
 - DatabaseSeeder to confirm my part
 - delete modal for donation
 - form tag on add.blade.php
 - delete modal for items to use common components
Delete
 - parameter on HomepageController
 - home.blade.php because the home has already created
 - parameter on item.add and route to homepage

Mavo39 committed:
[Controller/ItemController] Add and Edit some files
Changes:

Add
 - a function to pass the variable on HomeController
 - some statements on ItemController
 - 2 layouts setting on homepage.css
Edit
 - item / add.blade.php to use "selected" and change route to modal
 - route to add parameter on item.add
 - Seeder files to register to my database
Delete
 - a return view() on HomepageController.
It was to confirm my page and temporary.

Mavo39 committed:
Merge branch 'main' into Controller/ItemController

Mavo39 committed:
[Controller/ItemController] Edit some changes
Changes:

Add comment which one to use finally
Edit DonationController to adjust other member's code

Mavo39 committed:
[Controller/ItemController] Edit some changes
Changes:

Add comment which relationship to use finally
Edit CategorySeeder for other member's setting
Edit button tag to a tag to use parameter
Edit route to organize for member's setting
Delete duplicate on item model

kredo-school commented on app/Models/Category.php:
```diff
@@ -13,6 +13,7 @@ class Category extends Model
     protected $fillable = ['name', 'user_id'];
 
+    // need to decide to use "item" or "items"
     public function item()
```
redundant return $this->hasMany(Item::class); relationship

kredo-school commented on app/Models/Item.php:
```diff
@@ -12,20 +12,27 @@ class Item extends Model
 
+    public function categories()
+    {
+        $this->belongsTo(Category::class);
```
redundant $this->belongsTo(Category::class); relationship

kredo-school commented on app/Models/User.php:
```diff
@@ -59,19 +59,25 @@ public function favoriteItems() {
 
+    public function categories()
+    {
+        $this->hasMany(Category::class);
```
check this also

kredo-school commented on app/Http/Controllers/HomeController.php:
```diff
@@ -16,12 +17,18 @@ public function index(){
         return view('auth.passwords.password_reset_success');
     }
+
+    public function passVariable()
```
what's the use of this? i can't see a route in the web.php

Mavo39 committed:
[Controller/ItemController] - Modify some changes
Changes:

Edit ItemController to show selected category
Edit relation and Delete similar name relation
Edit item to items
Edit delete modal on Donation item to activate
Delete passVariable function from HomeController
Delete category parameter on app.blade.php
Delete comments from add / edit blade.php
Delete a parameter on route
